webapi/CopilotChat/Skills/ChatSkills/ChatSkill.cs

```diff
@@ -660,6 +695,35 @@ private int GetChatContextTokenLimit(string audience, string userIntent)
         return remainingToken;
     }
 
+    /// <summary>
+    /// Calculates total token usage of prompt chat completion and dependencies
+    /// </summary>
+    /// <param name="chatId">The chat ID</param>
+    /// <param name="userId">The user ID</param>
+    /// <param name="response">Message result of chat completion</param>
+    /// <param name="chatContext">Context maintained during response generation.</param>
+    private TokenUsage CalculateTokenUsage(string chatId, string userId, string response, SKContext chatContext)
+    {
+        var prompt = chatContext.Variables.TryGetValue("prompt", out string? renderedPrompt)
+            ? renderedPrompt
+            : string.Empty;
+
+        // Token usage of chat completion = prompt + completion
+        var chatCompletionTokenUsage = Utilities.TokenCount(prompt) + Utilities.TokenCount(response);
```

Comment:
this doesn't seem right. we're showing this as the token usage of the "Prompt" in the UX, but here we include the tokens of the completion as well. look at OAI's interface for [CompletionsUsage](https://learn.microsoft.com/en-us/dotnet/api/azure.ai.openai.completionsusage?view=azure-dotnet-preview), I would expect that we show "Prompt" and "Completion" token usage separately?

Reply:
The token usage shown in the UI is the total token usage for the prompt completion. Essentially with any completion request, total token usage = tokens in the prompt + tokens in the response. In the image below, the first "Hi! Can you help me plan a trip to Hawaii?" is considered the prompt (call) and the green part is the response (completion).
<img width="671" alt="image" src="https://github.com/microsoft/chat-copilot/assets/125500434/1a1fb416-ae9f-4ae6-a5d3-1175b3ee9903">

Since the chat streaming changes weren't backwards compatible with the token usage implementation in the feature branch, there isn't an easy way to get the total token usage from the function result as we can when just calling the chatCompletion function, so I had to calculate it manually like this. 

I previously aligned with John that we should just show the total usage (rather than splitting it up into prompt and completion for dependencies and bot response because this can get tricky to communicate well in the UI. 

I changed the copy in the info bubble to define this better
<img width="279" alt="image" src="https://github.com/microsoft/chat-copilot/assets/125500434/38387469-205d-45b0-aa29-7b68e7358874">

Reply:
> I previously aligned with John that we should just show the total usage (rather than splitting it up into prompt and completion for dependencies and bot response because this can get tricky to communicate well in the UI.

what happened after this? I'm guessing John is on board with these changes?

> I changed the copy in the info bubble to define this better

for `"Prompt token usage"`, I think that `"the total number of tokens used in the completion of the bot generation prompt"` is correct, but we also should mention that it includes the number of tokens from the actual prompt as well, since right now we only mention the "completion of" the prompt. 

I'm starting to see John's point that this can be tricky to show 😄.

really tho, I think we should be more explicit in the naming here to not confuse users with OAI's definition of these terms. which for prompt token usage, [only includes](https://learn.microsoft.com/en-us/dotnet/api/azure.ai.openai.completionsusage.prompttokens?view=azure-dotnet-preview#azure-ai-openai-completionsusage-prompttokens) the `"tokens in the provided prompts for the completion request"`, and not those in the completion itself. 

I'm thinking we make the following changes, let me know what you think:
1) show `"Prompt generation token usage"` as the `"total number of tokens used in the generation of the prompt and its final output"` (which is the sum of `TotalTokens` from the dependency functions, and the token count from the prompt).
2) show "Completion token usage" as the `"total number of tokens used in the completion of the response"`. 
3 and potentially show "Total token usage" as the sum of these two.

Reply:
Consulted with expert @matthewbolanos and we decided to get rid of the "Dependencies" bucket altogether and instead, be more specific with the token usage and split it up by explicit completions:

1. Extract audience tokens
1. Extract user intent tokens
1. Bot response tokens
1. Generate Semantic Memory tokens

Each of these counts will be total token usage (prompt + completion), so this will help remove the ambiguity of an overloaded "prompt" term